lib/AST/DeclContext.cpp

```diff
@@ -518,7 +518,10 @@ ResilienceExpansion DeclContext::getResilienceExpansion() const {
       // we serialize the parent's body.
       if (AFD->getDeclContext()->isLocalContext())
         continue;
-      
+
+      if (AFD->isInvalid())
```

Comment:
Oh, I think this is caused by the parser short-circuiting decl validation for constructors, subscripts and destructors outside of type context.

This is why I don't like 'ErrorType' or 'isInvalid'. We should always fully validate decls and set up invariants.